packages/react-instantsearch-hooks-web/src/widgets/Stats.tsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { useStats } from 'react-instantsearch-hooks';
+
+import { Stats as StatsUiComponent } from '../ui/Stats';
+
+import type {
+  StatsProps as StatsUiComponentProps,
+  StatsTranslationOptions,
+} from '../ui/Stats';
+import type { UseStatsProps } from 'react-instantsearch-hooks';
+
+type UiProps = Pick<
+  StatsUiComponentProps,
+  | 'nbHits'
+  | 'nbSortedHits'
+  | 'processingTimeMS'
+  | 'areHitsSorted'
+  | 'translations'
+>;
+
+export type StatsProps = Omit<StatsUiComponentProps, keyof UiProps> &
+  UseStatsProps & { translations?: Partial<UiProps['translations']> };
+
+export function Stats(props: StatsProps) {
+  const { nbHits, nbSortedHits, processingTimeMS, areHitsSorted } = useStats(
+    undefined,
+    { $$widgetType: 'ais.stats' }
+  );
+
+  const statsTranslation = (options: StatsTranslationOptions): string =>
+    options.areHitsSorted
+      ? `${options.nbSortedHits!.toLocaleString()} relevant results sorted out of ${options.nbHits.toLocaleString()} found in ${options.processingTimeMS.toLocaleString()}ms`
+      : `${options.nbHits.toLocaleString()} results found in ${options.processingTimeMS.toLocaleString()}ms`;
+
+  const uiProps: UiProps = {
+    nbHits,
+    nbSortedHits,
+    processingTimeMS,
+    areHitsSorted,
+    translations: {
+      stats: statsTranslation,
+      ...(props.translations || {}),
```

Comment:
I think with this implementation you override the translations twice if they're passed by the user. In other implementations it's like this:

```suggestion
export function Stats({ translations, ...props }: StatsProps) {
  const { nbHits, nbSortedHits, processingTimeMS, areHitsSorted } = useStats(
    undefined,
    { $$widgetType: 'ais.stats' }
  );

  const statsTranslation = (options: StatsTranslationOptions): string =>
    options.areHitsSorted
      ? `${options.nbSortedHits!.toLocaleString()} relevant results sorted out of ${options.nbHits.toLocaleString()} found in ${options.processingTimeMS.toLocaleString()}ms`
      : `${options.nbHits.toLocaleString()} results found in ${options.processingTimeMS.toLocaleString()}ms`;

  const uiProps: UiProps = {
    nbHits,
    nbSortedHits,
    processingTimeMS,
    areHitsSorted,
    translations: {
      stats: statsTranslation,
      ...translations,
```